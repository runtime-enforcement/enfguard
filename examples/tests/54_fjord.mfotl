ALWAYS (
  FORALL a, p.
    (dump_water(p)
    AND
    (a <- AVG(m; p; ONCE[0,60] measurement(p, "ph", m))))
    IMPLIES
    a >= 4.
)