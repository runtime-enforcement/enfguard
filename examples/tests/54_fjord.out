The formula
 □[0s,∞) (∀a:float. ∀p:string. (a:float <- AVG(m:float; p:string; ⧫[0s,60s] measurement(p:string, "ph", m:float))) ∧ dump_water(p:string) → geq(a:float, 4.) = 1)
is enforceable and types to
 □[0s,∞) ((∀a:float. (∀p:string. (((dump_water(p:string) : Sup) ∧:L (a:float <- AVG(m:float; p:string; ⧫[0s,60s] measurement(p:string, "ph", m:float))) : Sup) →:L geq(a:float, 4.) = 1 : Cau) : Cau)) : Cau) : Cau
[Enforcer] @1 OK.
[Enforcer] @1 nothing to do proactively.
[Enforcer] @2 OK.
[Enforcer] @2 nothing to do proactively.
[Enforcer] @3 reactively commands:
Suppress:
dump_water("place1")
OK.
[Enforcer] @3 OK.
